File: Linguaggi/Simulazioni/40/Python/Auto.py

from Veicolo import Veicolo

class Auto(Veicolo):
    
    def __init__(self, codice, targa, modello, marca, costo, cilindrata, capienza):
        super().__init__(codice, targa, modello, marca, costo)
        self.cilindrata = cilindrata
        self.capienza = capienza
        
    def __str__(self):
        return "automobile" + "\t" + super().__str__() + str(self.cilindrata)+"\t"+str(self.capienza) + "\t-\t-\t-"
        

File: Linguaggi/Simulazioni/40/Java/Auto.java
public class Auto extends Veicolo {
    protected int cilindrata;
    protected double capienza;

    public Auto(int codice, String targa, String modello, String marca, double costo, int cilindrata, double capienza) {
        super(codice, targa, modello, marca, costo);
        this.cilindrata = cilindrata;
        this.capienza = capienza;
    }

    public String toString() {
        return "automobile\t"
               +super.toString()+"\t"
               +this.cilindrata+"\t"
               +this.capienza+"\t-\t-\t-\t";
    }
}
